pkg/cmd/pipelinerun/describe_test.go

```diff
@@ -1212,3 +1212,122 @@ func TestPipelineRunDescribeV1beta1_taskrun_with_no_status(t *testing.T) {
 	}
 	golden.Assert(t, got, fmt.Sprintf("%s.golden", t.Name()))
 }
+
+func TestPipelineRunDescribe_last(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+
+	trs := []*v1alpha1.TaskRun{
+		tb.TaskRun("tr-1", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(2*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(5*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-2", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-3", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+	}
+
+	/*
```

Comment:
This comment isn't necessary. This test data will be sorted by the command run in this test, so it will be ordered by start time regardless of how elements are added to the PipelineRun list.

Reply:
I think perhaps I misunderstood this.
I added `cb.PipelineRunCreationTimestamp(clock.Now())` in the first one and `cb.PipelineRunCreationTimestamp(clock.Now().Add(5*time.Minute))` in the second in an impression that it will have 5 minutes later time for the second. When I changed the order the test failed which means they were not ordered. Any pointers from your side?

Reply:
It's because you are not updating the start time but the time stamp: tb.PipelineRunStartTime(clock.Now().Add(5*time.Minute)). Also, `Add` moves the start time forward with what you are doing such that pipeline-run2 would actually be starting five minutes earlier than pipeline-run. Below is how you would resolve this:

```go
pipelineRuns := []*v1alpha1.PipelineRun{
		tb.PipelineRun("pipeline-run2", "ns",
			cb.PipelineRunCreationTimestamp(clock.Now().Add(-5*time.Minute)),
			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
			tb.PipelineRunSpec("pipeline"),
			tb.PipelineRunStatus(
				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
					PipelineTaskName: "t-1",
					Status:           &trs[0].Status,
				}),
				tb.PipelineRunTaskRunsStatus("tr-3", &v1alpha1.PipelineRunTaskRunStatus{
					PipelineTaskName: "t-3",
					Status:           &trs[1].Status,
				}),
				tb.PipelineRunStatusCondition(apis.Condition{
					Status: corev1.ConditionTrue,
					Reason: resources.ReasonSucceeded,
				}),
				tb.PipelineRunStartTime(clock.Now().Add(-5*time.Minute)),
				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
			),
		),
		tb.PipelineRun("pipeline-run", "ns",
			cb.PipelineRunCreationTimestamp(clock.Now()),
			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
			tb.PipelineRunSpec("pipeline"),
			tb.PipelineRunStatus(
				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
					PipelineTaskName: "t-1",
					Status:           &trs[0].Status,
				}),
				tb.PipelineRunTaskRunsStatus("tr-2", &v1alpha1.PipelineRunTaskRunStatus{
					PipelineTaskName: "t-2",
					Status:           &trs[1].Status,
				}),
				tb.PipelineRunStatusCondition(apis.Condition{
					Status: corev1.ConditionTrue,
					Reason: resources.ReasonSucceeded,
				}),
				tb.PipelineRunStartTime(clock.Now()),
				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
			),
		),
	}
```

It's confusing, and this is one of many reasons test builders are being phased out.

Reply:
yaa you are right, I have added correct time (I guess) and it actually gets sorted ..